trl/trainer/utils.py

```
@@ -1492,3 +1497,37 @@ def generate_model_card(
         tokenizers_version=version("tokenizers"),
     )
     return card
+
+
+def get_comet_experiment_url() -> Optional[str]:
+    """If Comet integration is enabled returns the URL of the current Comet experiment
+    or None if disabled or no Comet experiment is currently running."""
+    if not is_comet_available():
+        return None
+
+    import comet_ml
+
+    if comet_ml.get_running_experiment() is not None:
+        return comet_ml.get_running_experiment().url
+
+    return None
+
+
+def log_table_to_comet_experiment(name: str, table: pd.DataFrame) -> None:
+    """
+    If Comet integration is enabled logs a table to the Comet experiment if it is currently running.
+
+    Args:
+        name (`str`):
+            Table name.
+        table (`pd.DataFrame`):
+            The Pandas DataFrame containing the table to log.
+    """
+    if not is_comet_available():
+        return
+
+    import comet_ml
```

Instead, I'd return an error here

```python
if is_comet_available():
    import comet_ml

[...]

def log_table_to_comet_experiment(name: str, table: pd.DataFrame):
    if not is_comet_available():
        raise ModuleNotFoundError("...")
```

---

Done